web/groups.go

```diff
@@ -142,46 +143,40 @@ func (s *AutograderService) updateGroup(ctx context.Context, sc scm.SCM, request
 		return err
 	}
 
-	group.Status = pb.Group_APPROVED
-	group.Users = users
-
-	org, err := sc.GetOrganization(ctx, &scm.GetOrgOptions{ID: course.OrganizationID})
-	if err != nil {
-		return fmt.Errorf("updateGroup: organization not found: %w", err)
-	}
-
 	if len(repos) == 0 {
-		// found no repos for the group; create group repo and team
-		if request.GetName() != "" {
+
+		if request.Name != "" && group.TeamID < 1 {
 			group.Name = request.Name
 		}
 
-		repo, team, err := createRepoAndTeam(ctx, sc, org, group.Name, group.Name, group.UserNames())
+		repo, team, err := createRepoAndTeam(ctx, sc, course.GetOrganizationID(), group.Name, request.UserNames())
 		if err != nil {
 			return err
 		}
-		// create database entry for group repository
-		groupRepo := &pb.Repository{
-			OrganizationID: course.OrganizationID,
-			RepositoryID:   repo.ID,
-			GroupID:        group.ID,
-			HTMLURL:        repo.WebURL,
-			RepoType:       pb.Repository_GROUP,
-		}
-		s.logger.Debugf("Creating group repo in the database with query: %+v", groupRepo)
-		if err := s.db.CreateRepository(groupRepo); err != nil {
+		repo.GroupID = group.GetID()
+		s.logger.Debugf("Creating group repo in the database: %+v", repo)
+		if err := s.db.CreateRepository(repo); err != nil {
 			return err
 		}
 		group.TeamID = team.ID
-	} else {
-		// github team already exists, update its members
-		// use the group's existing team ID obtained from the database above.
-		if err := updateGroupTeam(ctx, sc, org, group); err != nil {
+		// if updating group with existing team, group name will not be changed
+		// to avoid a mismatch between database and github names
+		s.logger.Debugf("updateGroup: name of the new GitHub team: %s, requested group name: %s", team.Name, request.Name)
+		if team.Name != request.Name {
+			group.Name = team.Name
+		}
+	}
+
+	// if there are changes in group members, update GitHub team
+	if group.ContainsAll(request) {
```

Comment:
This comment and the check itself appears to be contradicting. Should this be `!group.ContainsAll(request)`?

Reply:
We should augment the test case with a scenario that capture this.

Reply:
Oops, how come did it ever managed to correctly update group memberships before?..